### ISSUE TITLE
Updating scenes from first two books to work with MonteCarlo and PDF

### ISSUE THREAD

**Issue by dokotor** [RayTracing/raytracing.github.io#1544]
Hi, great work on all of the books. I have just finished reading them all and realised that you no longer can run the examples from previous books with the latest renderer version supplied with TheRestOfYourLife. The main reason is that it requires a list of lights to be passed alongside the world hittable.

Empty hittable list does not work, since pdf_mixed crashes when accessing its content. I also tried creating an infinite light sphere and thought that would be enough, but all of the shapes appear black.

Any idea how can I adapt previous scenes to the final renderer version?

**Comment by hollasch** [comment]
Are you running v4.0.0-alpha.2? That version has the fix for the list of lights, I believe. Otherwise, we're currently focusing on getting book 3 finished for the v4 update in the next several months, though there's lots of work still to be done. You can pick up the very latest changes in the `dev` branch.

**Comment by dokotor** [comment]
I don't think that I have made myself clear. In the third book the camera class accepts a list of hittable geometries and a list of hittable lights. Some of the previous examples (from books 1 and 2) didn't require any light in the scene. I am trying to understand how to run those examples with the final version of the renderer.

**Comment by hollasch** [comment]
Ah, I see. I don't think we really intended there to be a single renderer that does all of the scenes from all three books. Rather, there are three renderers, one for each book. It's worth thinking about, though.

**Comment by hollasch** [comment]
I'll keep this idea on the backlog.

**Comment by dokotor** [comment]
I think it would be somehow enough to implement a distant infinite light source and add it to the previous lightless scenes?
